package/MDAnalysis/analysis/density.py

```diff
@@ -180,11 +177,259 @@
 from .. import units
 from ..lib import distances
 from MDAnalysis.lib.log import ProgressMeter
+from MDAnalysis.lib.util import deprecate
+
+from .base import AnalysisBase
 
 import logging
 
 logger = logging.getLogger("MDAnalysis.analysis.density")
 
+class DensityAnalysis(AnalysisBase):
+    r"""Volumetric density analysis.
+
+    The trajectory is read, frame by frame, and the atoms in `atomgroup` are
+    histogrammed on a 3D grid with spacing `delta`.
+
+    Parameters
+    ----------
+    atomgroup : AtomGroup or UpdatingAtomGroup
+            Group of atoms (such as all the water oxygen atoms) being analyzed.
+            This can be an :class:`~MDAnalysis.core.groups.UpdatingAtomGroup` for
+            selections that change every time step.
+    delta : float (optional)
+            Bin size for the density grid in ångström (same in x,y,z).
+            Slice the trajectory as "trajectory[start:stop:step]"; default
+            is to read the whole trajectory.
+    metadata : dict (optional)
+            `dict` of additional data to be saved with the object; the meta
+            data are passed through as they are.
+    padding : float (optional)
```

Comment:
Should the default arguments of kwargs be noted in the docs?

Reply:
If they are explicit in the call signature (i.e., we did not use `**kwargs`) then they are visible in the generated docs and with `help(...)`. I don't think we have an explicit policy but there's something to be said for being DRY.

Do you prefer them duplicated in the docs?

Reply:
I think so long as there is an explicit signature, they don't need to be there (and protects against changing the defaults in future versions) -- that being said I have been putting default values at the end of the explanatory part, [following examples in PEP 257](https://www.python.org/dev/peps/pep-0257/). I was just wondering if you had a specific policy on it.

Reply:
We don't have an explicit policy in https://github.com/MDAnalysis/mdanalysis/wiki/WritingDocumentation. I vaguely remember discussion with @kain88-de where he convinced me that we should avoid `**kwargs` and use explicit kwargs because they immediately expose defaults, which reduces the need to list them explicitly.